src/pages/auth/login.vue

```diff
@@ -80,21 +105,41 @@ const loading = ref(false);
 const data = reactive<LoginData>({
   email: '',
   password: '',
+  token: '',
 });
 
 const hasCredentialError = ref(false);
+const hasMFAEnabled = ref(false);
+const hasWrongMFAToken = ref(false);
+
+let emailBefore = '';
 
 async function submitLogin() {
   loading.value = true;
   hasCredentialError.value = false;
+  hasWrongMFAToken.value = false;
+
+  // Only reset MFA if email changed
+  if (emailBefore !== data.email) {
```

Comment:
I think this could be a `watch` so it resets when the user changes the input and you don't need the `emailBefore` variable, maybe something like
 
 ```ts
 watch(() => data.email, () => (hasMFAEnabled.value = false));
 ```